src/frontend.rs

```diff
@@ -550,17 +739,13 @@ pub mod tests {
     use super::*;
     use crate::backends::mock_signed::MockSigner;
 
-    macro_rules! args {
-        ($($arg:expr),+) => {vec![$(OperationArg::from($arg)),*]}
-    }
-
     macro_rules! op {
-        (eq, $($arg:expr),+) => { Operation(NativeOperation::EqualFromEntries, args!($($arg),*)) };
-        (ne, $($arg:expr),+) => { Operation(NativeOperation::NotEqualFromEntries, args!($($arg),*)) };
-        (gt, $($arg:expr),+) => { Operation(NativeOperation::GtFromEntries, args!($($arg),*)) };
-        (lt, $($arg:expr),+) => { Operation(NativeOperation::LtFromEntries, args!($($arg),*)) };
-        (contains, $($arg:expr),+) => { Operation(NativeOperation::ContainsFromEntries, args!($($arg),*)) };
-        (not_contains, $($arg:expr),+) => { Operation(NativeOperation::NotContainsFromEntries, args!($($arg),*)) };
+        (eq, $($arg:expr),+) => { Operation::EqualFromEntries($(OperationArg::try_from($arg).unwrap()),*) };
```

Comment:
This is cool, now the compiler will complain when using the macro with the wrong number of arguments because it's building an enum.